ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java

```diff
@@ -326,26 +328,47 @@ public SignedBeaconBlock deserializeSignedBeaconBlock(final Bytes serializedSign
   }
 
   public SignedBlockContainer deserializeSignedBlockContainer(
-      final Bytes serializedSignedBlockContainer) {
-    final UInt64 slot =
-        BeaconBlockInvariants.extractSignedBlockContainerSlot(serializedSignedBlockContainer);
-    return atSlot(slot)
-        .getSchemaDefinitions()
+      final Bytes serializedSignedBlockContainer, final Optional<String> milestone) {
+
+    final SchemaDefinitions schemaDefinition =
+        getSchemaDefinitionsForMilestone(milestone)
+            .orElseGet(getSchemaDefinitionForSignedBlockSSZ(serializedSignedBlockContainer));
+
+    return schemaDefinition
         .getSignedBlockContainerSchema()
         .sszDeserialize(serializedSignedBlockContainer);
   }
 
   public SignedBlockContainer deserializeSignedBlindedBlockContainer(
-      final Bytes serializedSignedBlindedBlockContainer) {
-    final UInt64 slot =
-        BeaconBlockInvariants.extractSignedBlockContainerSlot(
-            serializedSignedBlindedBlockContainer);
-    return atSlot(slot)
-        .getSchemaDefinitions()
+      final Bytes serializedSignedBlindedBlockContainer, final Optional<String> milestone) {
+
+    final SchemaDefinitions schemaDefinition =
+        getSchemaDefinitionsForMilestone(milestone)
+            .orElseGet(getSchemaDefinitionForSignedBlockSSZ(serializedSignedBlindedBlockContainer));
+
+    return schemaDefinition
         .getSignedBlindedBlockContainerSchema()
         .sszDeserialize(serializedSignedBlindedBlockContainer);
   }
 
+  private Optional<SchemaDefinitions> getSchemaDefinitionsForMilestone(
+      final Optional<String> milestone) {
+    return milestone
+        .map(v -> SpecMilestone.valueOf(v.toUpperCase(Locale.ROOT)))
```

Comment:
I think it would be useful to have a static method in SpecMilestone like:

```java
public static SpecMilestone fromEthConsensusVersionHeader(final String ethConsensusVersion) {
    return SpecMilestone.valueOf(ethConsensusVersion.toUpperCase(Locale.ROOT));
}
```

Could be reused in tests, etc.